src/pudl/transform/eia860m.py

```diff
@@ -56,8 +58,27 @@ def core_eia860m__changelog_generators(
         raw_eia860__generator=pd.DataFrame(
             columns=list(eia860_columns)
         ).convert_dtypes(),
-    )[
-        # drop all the non 860m cols
+    ).assign(
+        # In order to be able to compare the values in this table to those reported
+        # elsewhere, we need to translate these categories to the associated codes, as
+        # the strings associated with the codes vary from table to table. See the
+        # core_eia__codes_sector_consolidated table for the canonical definition of the
+        # codes.
+        sector_id_eia=lambda df: df["sector_name_eia"].map(
+            {
+                "Electric Utility": 1,
+                "IPP Non-CHP": 2,
+                "IPP CHP": 3,
+                "Commercial Non-CHP": 4,
+                "Commercial CHP": 5,
+                "Industrial Non-CHP": 6,
+                "Industrial CHP": 7,
+            }
+        )
+    )
```

Comment:
I added this translation early, because of the column removal that happens below.

We don't store a mapping of sector names to sector codes anywhere else, because the names are not consistently used across tables, and everywhere else the code is directly reported.  Really we should probably replace the sector names everywhere else with our human-readable labels (or just drop them from the core tables, and merge in the human readable labels for output tables)

Reply:
dang ya it would obviously be better to make an encoder table and use it across the board but that feels very oos for this pr so this seems fine.

Reply:
I had to physically restrain myself...

Reply:
good jerb